pkg/controllers/provisioning/provisioner.go

```diff
@@ -120,45 +109,100 @@ func (p *Provisioner) provision(ctx context.Context) error {
 	return nil
 }
 
-// isProvisionable ensure that the pod can still be provisioned.
-// This check is needed to prevent duplicate binds when a pod is scheduled to a node
-// between the time it was ingested into the scheduler and the time it is included
-// in a provisioner batch.
-func (p *Provisioner) isProvisionable(ctx context.Context, candidate *v1.Pod) (bool, error) {
-	stored := &v1.Pod{}
-	if err := p.kubeClient.Get(ctx, client.ObjectKeyFromObject(candidate), stored); err != nil {
-		if errors.IsNotFound(err) {
-			return false, nil
+func (p *Provisioner) getPods(ctx context.Context) ([]*v1.Pod, error) {
+	var podList v1.PodList
+	if err := p.kubeClient.List(ctx, &podList, client.MatchingFields{"spec.nodeName": ""}); err != nil {
+		return nil, fmt.Errorf("listing pods, %w", err)
+	}
+	var pods []*v1.Pod
+	for i := range podList.Items {
+		if isProvisionable(&podList.Items[i]) {
+			pods = append(pods, &podList.Items[i])
```

Comment:
```suggestion
			pods = append(pods, podList.Items[i].DeepCopy())
```

We relax the pods later, if we deep copy here we ensure that we don't mess the list cache up.

Reply:
IIUC it's a copy -- may want to rethink if we care about memory at some point.